clients/data_types.go

```diff
@@ -131,3 +148,8 @@ type CreateBaremetalInfo struct {
 	BaremetalSpecs []CloudProperty `json:"baremetal_specs"`
 	Deployment     string          `json:"deployment"`
 }
+
+// deployment
+type Deployment struct {
+	Name string `yaml:"name"`
```

Comment:
yeah, change to json, you are not expecting YAML content which is what using yaml will do.